weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/interface.ts

```diff
@@ -0,0 +1,501 @@
+      : span.name,
+    opVersionRef: span.name.startsWith(WANDB_ARTIFACT_REF_PREFIX)
+      ? span.name
+      : null,
+  };
+};
+
+export const useCall = (key: CallKey): Loadable<CallSchema | null> => {
+  const calls = useRuns(
+    {
+      entityName: key.entity,
+      projectName: key.project,
+      streamName: PROJECT_CALL_STREAM_NAME,
+    },
+    {
+      callIds: [key.callId],
+    }
+  );
+
+  return useMemo(() => {
+    const callResult = calls.result?.[0] ?? null;
+    const result = callResult
+      ? spanToCallSchema(key.entity, key.project, callResult)
+      : null;
+    if (calls.loading) {
+      return {
+        loading: true,
+        result,
+      };
+    } else {
+      return {
+        loading: false,
+        result,
+      };
+    }
+  }, [calls.loading, calls.result, key.entity, key.project]);
+};
+
+type CallFilter = {
+  // Filters are ANDed across the fields and ORed within the fields
+  // Commented out means not yet implemented
+  //   opVersionRefs?: string[];
+  inputObjectVersionRefs?: string[];
+  outputObjectVersionRefs?: string[];
+  //   traceIds?: string[];
+  parentIds?: string[];
+  //   callIds?: string[];
+  //   traceRootsOnly?: boolean;
+};
+export const useCalls = (
+  entity: string,
+  project: string,
+  filter: CallFilter
+): Loadable<CallSchema[]> => {
+  const calls = useRuns(
+    {
+      entityName: entity,
+      projectName: project,
+      streamName: PROJECT_CALL_STREAM_NAME,
+    },
+    {
+      // opUris?: string[];
+      inputUris: filter.inputObjectVersionRefs,
+      outputUris: filter.outputObjectVersionRefs,
+      // traceId?: string;
+      parentIds: filter.parentIds,
+      // traceRootsOnly?: boolean;
+      // callIds?: string[];
+    }
+  );
+  return useMemo(() => {
+    const result = (calls.result ?? []).map(run =>
+      spanToCallSchema(entity, project, run)
+    );
+    if (calls.loading) {
+      return {
+        loading: true,
+        result,
+      };
+    } else {
+      return {
+        loading: false,
+        result,
+      };
+    }
+  }, [entity, project, calls.loading, calls.result]);
+};
+
+type OpVersionKey = {
+  entity: string;
+  project: string;
+  opId: string;
+  versionHash: string;
+};
+
+export const refUriToOpVersionKey = (refUri: RefUri): OpVersionKey => {
+  const refDict = refStringToRefDict(refUri);
+  if (
+    refDict.filePathParts.length !== 1 ||
+    refDict.refExtraTuples.length !== 0 ||
+    refDict.filePathParts[0] !== 'obj'
+  ) {
+    throw new Error('Invalid refUri: ' + refUri);
+  }
+  return {
+    entity: refDict.entity,
+    project: refDict.project,
+    opId: refDict.artifactName,
+    versionHash: refDict.versionCommitHash,
+  };
+};
+export const opVersionKeyToRefUri = (key: OpVersionKey): RefUri => {
+  return `wandb-artifact:///${key.entity}/${key.project}/${key.opId}:${key.versionHash}/obj`;
+};
+
+type MVPOpCategory = 'train' | 'predict' | 'score' | 'evaluate' | 'tune';
```

Comment:
I'd probably get rid of the `MVP` prefix, better to convey that with a comment I think.

Reply:
👍